fuzz/elrond-ethereum-bridge/fuzzElrondEthereumBridge_test.go

```diff
@@ -0,0 +1,390 @@
+	return arwenTestRoot
+}
+
+func newExecutorWithPaths() *fuzzExecutor {
+	fileResolver := mc.NewDefaultFileResolver().
+		ReplacePath(
+			"price-aggregator.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/price-aggregator/price-aggregator.wasm")).
+		ReplacePath(
+			"multisig.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/multisig/multisig.wasm")).
+		ReplacePath(
+			"egld-esdt-swap.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/egld-esdt-swap/egld-esdt-swap.wasm")).
+		ReplacePath(
+			"esdt-safe.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/esdt-safe/esdt-safe.wasm")).
+		ReplacePath(
+			"multi-transfer-esdt.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/multi-transfer-esdt/multi-transfer-esdt.wasm")).
+		ReplacePath(
+			"ethereum-fee-prepay.wasm",
+			filepath.Join(getTestRoot(), "elrond-ethereum-bridge/ethereum-fee-prepay/ethereum-fee-prepay.wasm"))
+
+	fe, err := newFuzzExecutor(fileResolver)
+	if err != nil {
+		panic(err)
+	}
+	return fe
+}
+
+func TestElrondEthereumBridge(t *testing.T) {
+	if !*fuzz {
+		t.Skip("skipping test; only run with --fuzz argument")
+	}
+
+	fe := newExecutorWithPaths()
+	defer fe.saveGeneratedScenario()
+
+	err := fe.initData()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// TODO: Uncomment once aggregator is integrated
+	// The current version doesn't have relayer incentives and user fees
+
+	/*
+		err = fe.setupAggregator()
+		if err != nil {
+			t.Error(err)
+		}
+	*/
+
+	nrRelayers := 2
+	nrUsers := 2
+	initialBalance := big.NewInt(INIT_BALANCE)
+	err = fe.initAccounts(nrRelayers, nrUsers, initialBalance)
+	if err != nil {
+		t.Error(err)
+	}
+
+	multisigInitArgs := MultisigInitArgs{
+		requiredStake: big.NewInt(1000),
+		slashAmount:   big.NewInt(500),
+		quorum:        len(fe.data.actorAddresses.relayers) / 2,
+		boardMembers:  fe.data.actorAddresses.relayers,
+	}
+	err = fe.deployMultisig(&multisigInitArgs)
+	if err != nil {
+		t.Error(err)
+	}
+
+	deployChildContractsArgs := DeployChildContractsArgs{
+		egldEsdtSwapCodePath:      "file:egld-esdt-swap.wasm",
+		multiTransferEsdtCodePath: "file:multi-transfer-esdt.wasm",
+		ethereumFeePrepayCodePath: "file:ethereum-fee-prepay.wasm",
+		esdtSafeCodePath:          "file:esdt-safe.wasm",
+		priceAggregatorAddress:    "sc:price-aggregator",
+		wrappedEgldTokenId:        "str:WEGLD-123456",
+		wrappedEthTokenId:         "str:WETH-abcdef",
+		tokenWhitelist:            []string{},
+	}
+	err = fe.setupChildContracts(&deployChildContractsArgs)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var seed int64
+	if *seedFlag == 0 {
+		seed = time.Now().UnixNano()
+	} else {
+		seed = *seedFlag
+	}
+	fe.log("Random seed: %d\n", seed)
+	r := rand.New(rand.NewSource(seed))
+	r.Seed(seed)
+
+	fe.randSource = *r
+
+	re := fuzzutil.NewRandomEventProvider(r)
+	for stepIndex := 1; stepIndex <= *iterationsFlag; stepIndex++ {
+		re.Reset()
+
+		switch {
+		case re.WithProbability(0.2):
```

Comment:
Please consider using Roulette implementation. It's cleaner and more extensible. Like the case when you would want to add a bunch of other events, with the current re.WithProbability, you would have to change all current probabilities and make sure the new sum does not surpass 1. With Roulette this would not be necessary